js/src/components/paid-ads/asset-group/images-selector.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import GridiconPlusSmall from 'gridicons/dist/plus-small';
+import GridiconCrossCircle from 'gridicons/dist/cross-circle';
+
+/**
+ * Internal dependencies
+ */
+import AppButton from '.~/components/app-button';
+import useCroppedImageSelector from '.~/hooks/useCroppedImageSelector';
+import './images-selector.scss';
+
+/**
+ * @typedef {Object} AssetImageConfig
+ * @property {number} minWidth The minimum width.
+ * @property {number} minHeight The minimum height.
+ * @property {number} suggestedWidth The suggested width.
+ * @property {number} suggestedHeight The suggested height.
+ */
+
+/**
+ * Renders a selector for asset images.
+ *
+ * @param {Object} props React props.
+ * @param {AssetImageConfig} props.imageConfig The config of the asset image.
+ * @param {string[]} props.initialImageUrls The initial image URLs.
+ * @param {number} [props.maxNumberOfImages=0] The maximum number of images. 0 by default and it means unlimited number.
+ */
+export default function ImagesSelector( {
+	imageConfig,
+	initialImageUrls = [],
+	maxNumberOfImages = 0,
+} ) {
+	const [ replacingImage, setReplacingImage ] = useState( null );
+	const [ images, setImages ] = useState( () =>
+		// The asset images fetched from Google Ads are only URLs.
+		initialImageUrls.map( ( url ) => ( { url, id: url, alt: '' } ) )
+	);
+
+	const removeImage = ( deletedImage ) => {
+		setImages( images.filter( ( { id } ) => id !== deletedImage.id ) );
+	};
+
+	const handle = useCroppedImageSelector( {
+		...imageConfig,
+		onDelete: removeImage,
+		onSelect( image ) {
+			const nextImages = [ ...images ];
+
+			// Find if there is a duplicate image first.
+			let index = nextImages.findIndex( ( { id } ) => id === image.id );
+
+			if ( replacingImage ) {
```

Comment:
💅  I found `replacingImage` a bit confusing in order to identify if `replacingImage` was

A:  the new selected image to replace the previous one. 
B:  the previous image to be replaced by the new selected image. 

After inspecting all the code I guess its B. So I'd suggest replacing `image` with `newSelectedImage` and `previousSelectedImage`

Reply:
Consider that the `replacingImage` may not only be replaced but may also be swapped places or removed without further processing. I renamed it to `awaitingActionImage` in https://github.com/woocommerce/google-listings-and-ads/pull/1819/commits/84f63e919ad5d4e5a12d80475f3c73c560785beb.

Reply:
Thanks for the adjustment. I'm still not sure about the new name `awaitingActionImage`. The word `await` and `action` in the name could have ambiguous meanings and I found it confusing. 

I don't want to sound pushy about my suggestion. But what is against `newSelectedImage` and `previousSelectedImage`?  I think it clearly defines what the user is doing in the UI.

Reply:
> I'm still not sure about the new name `awaitingActionImage`. The word `await` and `action` in the name could have ambiguous meanings and I found it confusing.

It's precisely to avoid ambiguity, the word "selected" is used only in the `onSelect` callback and refers to "the selected image", and the word was avoided in the rest of `ImagesSelector`.

Otherwise, for example, would the `previousSelectedImage` literally mean the previously selected image in the `onSelect` callback, or the previously selected image in the callback of the replacement buttons?

The `awaitingActionImage` keeps the last image data sent from the associated click handler of a replacement button. After "the selected image" is called back from `onSelect`, the next action for `awaitingActionImage` could be:
- pointing out which image to be replaced with the selected image.
- pointing out which image to be swapped the position with the selected image.
- nothing if the reset action for `awaitingActionImage` has occurred in the `onDelete` callback.

Thus, the `awaitingActionImage` is an image **awaiting** something to happen, and when it does, it takes **action** according to the situation at the moment.

---

Besides, I didn't rename `image` to `newSelectedImage` because the `image` as the parameter to `onSelect` callback should be enough to indicate that it is the "selected image", and the `onSelect` implementation is only a few lines long, so I thought it's probably not confusing. But it's just my opinion. I'm happy to rename `image` to `selectedImage` if it's still confusing you.